config/v1/types_infrastructure.go

```diff
@@ -182,6 +182,30 @@ type BareMetalPlatformStatus struct {
 	// datacenter DNS, a DNS service is hosted as a static pod to serve those hostnames
 	// to the nodes in the cluster.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// provisioningInterface is the name of the network interface on a Baremetal server connected
+	// to the provisioning network.
+	ProvisioningInterface string `json:"provisioningInterface"`
+
+	// provisioningIP is the IP address assigned to the provisioningInterface for provisioning
+	// the baremetal node. This IP address should be within the provisioning subnet, and
+	// outside of the DHCP range.
+	ProvisioningIP string `json:"provisioningIP"`
+
+	// provisioningNetworkCIDR is the network on which the baremetal nodes are provisioned.
+	// The provisioningIP and the IPs in the dhcpRange all come from within this network.
+	ProvisioningNetworkCIDR string `json:"provisioningNetworkCIDR"`
+
+	// dhcpRange is the IP address range on the provisioning subnet from which baremetal hosts
+	// can be assigned an IP address.
+	DHCPRange string `json:"dhcpRange"`
```

Comment:
To make it clear what this is for should we call this ProvisioningDHCPRange?

Also I think a previous reviewer asked for clarification on the format here - currently we expect a comma-separated list of two IPs, so we should mention that in the comment.  

Alternatively we could instead have two parameters (ProvisioningDHCPStart and ProvisioningDHCPEnd), or a CIDR to express the range, I don't really have a strong opinion on which although it would be nice to decide soon so we can align the related installer PR with the same interface.